core/src/main/scala/com/evolutiongaming/kafka/flow/kafka/KafkaModule.scala

```diff
@@ -1,83 +1,80 @@
 package com.evolutiongaming.kafka.flow.kafka
 
-import cats.effect.Blocker
-import cats.effect.Clock
-import cats.effect.ConcurrentEffect
-import cats.effect.ContextShift
-import cats.effect.Resource
-import cats.effect.Timer
+import cats.effect.{Blocker, Clock, ConcurrentEffect, ContextShift, Resource, Timer}
 import cats.syntax.all._
-import com.evolutiongaming.catshelper.FromTry
-import com.evolutiongaming.catshelper.LogOf
-import com.evolutiongaming.catshelper.ToFuture
-import com.evolutiongaming.catshelper.ToTry
+import com.evolutiongaming.catshelper.CatsHelper._
+import com.evolutiongaming.catshelper.{FromTry, LogOf, ToFuture, ToTry}
 import com.evolutiongaming.kafka.flow.LogResource
-import com.evolutiongaming.kafka.journal.KafkaConfig
-import com.evolutiongaming.kafka.journal.KafkaHealthCheck
-import com.evolutiongaming.kafka.journal.RandomIdOf
 import com.evolutiongaming.kafka.journal.util.SkafkaHelper._
-import com.evolutiongaming.kafka.journal.{KafkaConsumerOf => JournalConsumerOf}
-import com.evolutiongaming.kafka.journal.{KafkaProducerOf => JournalProducerOf}
-import com.evolutiongaming.skafka.consumer.AutoOffsetReset
-import com.evolutiongaming.skafka.consumer.{ConsumerConfig, ConsumerMetrics, ConsumerOf => RawConsumerOf}
-import com.evolutiongaming.skafka.producer.{ProducerConfig, ProducerMetrics, ProducerOf => RawProducerOf}
-import com.evolutiongaming.smetrics.CollectorRegistry
-import com.evolutiongaming.smetrics.MeasureDuration
+import com.evolutiongaming.kafka.journal.{
+  KafkaConfig,
+  KafkaHealthCheck,
+  RandomIdOf,
+  KafkaConsumerOf => JournalConsumerOf,
+  KafkaProducerOf => JournalProducerOf
+}
+import com.evolutiongaming.skafka.consumer._
+import com.evolutiongaming.skafka.producer.{Producer, ProducerMetrics, ProducerOf}
+import com.evolutiongaming.smetrics.{CollectorRegistry, MeasureDuration}
+import monocle.macros.syntax.lens._
 import scodec.bits.ByteVector
 
 trait KafkaModule[F[_]] {
 
   def healthCheck: KafkaHealthCheck[F]
 
-  def consumerOf: ConsumerOf[F]
-  def producerOf: RawProducerOf[F]
+  def consumerOf: Resource[F, Consumer[F, String, ByteVector]]
+  def producerOf: Resource[F, Producer[F]]
 
 }
 object KafkaModule {
 
   def of[F[_]: ConcurrentEffect: ContextShift: FromTry: ToTry: ToFuture: Timer: LogOf](
     applicationId: String,
-    config: ConsumerConfig,
+    groupId: String,
+    config: KafkaConfig,
```

Comment:
It looks a bit ugly to pass the config and then change it in the code. The user might be surprised and annoyed when he or she tries to change part of the config, for it to only be changed in `KafkaModule` code. I know it was before your change.

What if we separate concerns and construct appropriate `KafkaConfig` in a separate class, and then just pass it here for further use?

Reply:
> What if we separate concerns and construct appropriate KafkaConfig in a separate class, and then just pass it here for further use?

Nice idea, let me try to implement it.